src/signal-forms/signal-input.directive.ts

```diff
@@ -13,10 +14,17 @@ import { FormField } from './form-field';
     '[class.control-pending]': 'this.formField.state() === "PENDING"',
   },
 })
-export class SignalInputDirective {
-  @Input() formField!: FormField<unknown>;
+export class SignalInputDirective implements OnInit, OnDestroy {
+  private _destroyed = new Subject<void>();
+  private _debounced = new Subject<unknown>();
+  @Input() formField!: FormField;
+  @Input() debounce: string | number | null = null;
```

Comment:
Would there be a better way to do this?
For example, with another directive that does this?
That why we can also introduce multiple behaviors (e.g. throttle) in isolated directives instead of this one.

Reply:
Yes! I actually like that a lot better than "polluting" this directive with additional, optional behavior!

I am trying to think through the problem we would need to solve...

I guess it would be triggering a custom function for the current `onModelChange` function (or allowing to override it) and enable the other directive to ultimately set the value through `this.formField.value.set(value)`.

Reply:
Tbh, I don't know what's the best way to do this.
A quick google search pointed multiple sources using a debounce on the keyup event, e.g.

https://jasonwhite.xyz/posts/2020/10/16/angular-creating-a-debounce-directives/